modules/exploits/windows/http/exchange_ecp_dlp_policy.rb

```diff
@@ -164,7 +164,7 @@ def owa_login
       fail_with(Failure::Unreachable, 'Failed to access OWA login page')
     end
 
-    unless res.code == 200 && cookie_jar.grep(/^cadata/).any?
+    unless res.code == 200 && cookie_jar.cookies.any? { |cookie| cookie.name.start_with?('cadata') }
```

Comment:
I'm not sure if the old code specifically wanted any cookie that started with `cadata`, but I decided to align with the previous semantics for consistency.

Note, I wasn't able to get a reverse shell for this module against Christophe's test environment - either on master or after this change. However I verified this line's boolean works as expected for the logged in/not logged in scenarios, so I think this is good to go :+1: